src/libraries/System.Collections/src/System/Collections/Generic/Queue.cs

```diff
@@ -386,6 +386,64 @@ public void TrimExcess()
             }
         }
 
+        /// <summary>
+        /// Ensures that the capacity of this Queue is at least the specified <paramref name="capacity"/>.
+        /// </summary>
+        /// <param name="capacity">The minimum capacity to ensure</param>
+        public int EnsureCapacity(int capacity)
+        {
+            if (capacity < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(capacity), capacity, SR.ArgumentOutOfRange_NeedNonNegNum);
+            }
+
+            if (_array.Length < capacity)
+            {
+                int newCapacity = (int)((long)(_array.Length == 0 ? 1 : _array.Length) * (long)GrowFactor / 100);
```

Comment:
Why set the capacity to `1` if the array is empy? FWIW we should probably just replicate the logic as defined in the Enqueue method: https://github.com/dotnet/runtime/blob/cb5431d0c0790a69925145a15952a9601b9be102/src/libraries/System.Collections/src/System/Collections/Generic/Queue.cs#L186-L191

Reply:
Is the outer cast to `long` necessary?

Reply:
Because we have introduced "continuously minimumGrow" logic as suggested for List, so if not set capacity to 1 for empty array in initial newCapacity, following loop will be dead loop.

Reply:
I'm not sure I understand, could you elaborate?

Reply:
For example, if array is empty:
int newcapacity = (int)(_array.Length * (long)GrowFactor / 100);  // newcapacity will be 0
So that 
while (newCapacity < capacity) //  make loop dead
{
                    newCapacity = (int)((long)newCapacity * (long)GrowFactor / 100);  // newcapacity will be 0 forever
}

Reply:
> int newcapacity = (int)(_array.Length * (long)GrowFactor / 100); // newcapacity will be 0

Couldn't you make this like the `List` implementation?

```csharp
int newCapacity = _array.Length == 0 ? MinimumGrow : (int)(_array.Length * (long)GrowFactor / 100);
```

Reply:
Fixed.